/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
syntax = "proto3";

package org.ffsf.demo.inventory;

option java_package = "org.ffsf.demo.inventory.generated";
option java_outer_classname = "ProtobufMessages";
option java_multiple_files = false;

// ---------------------------------------------------------------------
// Inventory messages
// ---------------------------------------------------------------------

message RestockItem {
    string item_id = 1;
    int32 quantity = 2;
}

message RequestItem {
    int32 quantity = 1;
}

message ItemAvailability {
    enum Status {
        IN_STOCK = 0;
        OUT_OF_STOCK = 1;
    }

    Status status = 1;
    int32 quantity = 2;
}
